langchain/src/prompts/base.ts

```diff
@@ -9,6 +9,7 @@ export type SerializedBasePromptTemplate = ReturnType<
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type InputValues = Record<string, any>;
+export type PartialValues = Record<string, string | (() => Promise<string>)>;
```

Comment:
I think we can also allow functions that return a string (not wrapped in a promise)